### ISSUE TITLE
Reproducing Sample Segmentations

### ISSUE THREAD

**Issue by StefanFischer** [primakov/DuneAI-Automated-detection-and-segmentation-of-non-small-cell-lung-cancer-computed-tomography-images#2]
Hello,

thank you for sharing your code and model parameters to the community. I am trying to reproduce your test sample results ("\Software for qualitative assesment\test_data"). For that i just run the Jupyter Notebook "Automatic segmentation script\Automatic segmentation script.ipynb" without any changes. But as an output i get moderate changes in the tumor segmentations for all three example patients (Pat1, Pat2, Pat3). Here are my results for Pat1, where DL_mask_github_pat1.nrrd is the sample file from Github and DL_mask_reproduced_pat1.nrrd is my reproduced segmentation (without any changes to your notebook pipeline). 

[DL_mask_pat1.zip](https://github.com/primakov/DuneAI-Automated-detection-and-segmentation-of-non-small-cell-lung-cancer-computed-tomography-images/files/11048776/DL_mask_pat1.zip)

**Comment by primakov** [comment]
Hello @StefanFischer ,

Pardon the late reaction.

Could you please share your environment config too?

**Comment by StefanFischer** [comment]
Hi Primakov,

thanks for your response. Here is my env configuration.

# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main  
_openmp_mutex             5.1                       1_gnu  
absl-py                   1.4.0                    pypi_0    pypi
acvl-utils                0.2                      pypi_0    pypi
anyio                     3.6.2                    pypi_0    pypi
argon2-cffi               21.3.0                   pypi_0    pypi
argon2-cffi-bindings      21.2.0                   pypi_0    pypi
argparse                  1.4.0                    pypi_0    pypi
astunparse                1.6.3                    pypi_0    pypi
attrs                     22.2.0                   pypi_0    pypi
backcall                  0.2.0                    pypi_0    pypi
batchgenerators           0.25                     pypi_0    pypi
beautifulsoup4            4.12.0                   pypi_0    pypi
bleach                    6.0.0                    pypi_0    pypi
ca-certificates           2022.12.7            ha878542_0    conda-forge
cachetools                5.3.0                    pypi_0    pypi
certifi                   2022.12.7          pyhd8ed1ab_0    conda-forge
cffi                      1.15.1                   pypi_0    pypi
charset-normalizer        3.1.0                    pypi_0    pypi
click                     8.1.3                    pypi_0    pypi
connected-components-3d   3.10.5                   pypi_0    pypi
cudatoolkit               11.2.2              hbe64b41_10    conda-forge
cudnn                     8.1.0.77             h90431f1_0    conda-forge
cycler                    0.11.0                   pypi_0    pypi
dataclasses               0.6                      pypi_0    pypi
debugpy                   1.6.6                    pypi_0    pypi
decorator                 5.1.1                    pypi_0    pypi
defusedxml                0.7.1                    pypi_0    pypi
dicom2nifti               2.4.8                    pypi_0    pypi
docopt                    0.6.2                    pypi_0    pypi
entrypoints               0.4                      pypi_0    pypi
fastjsonschema            2.16.3                   pypi_0    pypi
fastremap                 1.13.4                   pypi_0    pypi
fill-voids                2.0.3                    pypi_0    pypi
flatbuffers               23.3.3                   pypi_0    pypi
fonttools                 4.38.0                   pypi_0    pypi
future                    0.18.3                   pypi_0    pypi
gast                      0.4.0                    pypi_0    pypi
ghp-import                2.1.0                    pypi_0    pypi
google-auth               2.16.2                   pypi_0    pypi
google-auth-oauthlib      0.4.6                    pypi_0    pypi
google-pasta              0.2.0                    pypi_0    pypi
grpcio                    1.51.3                   pypi_0    pypi
h5py                      3.8.0                    pypi_0    pypi
idna                      3.4                      pypi_0    pypi
imageio                   2.26.1                   pypi_0    pypi
importlib-metadata        6.1.0                    pypi_0    pypi
importlib-resources       5.12.0                   pypi_0    pypi
ipykernel                 6.16.2                   pypi_0    pypi
ipython                   7.34.0                   pypi_0    pypi
ipython-genutils          0.2.0                    pypi_0    pypi
ipywidgets                8.0.5                    pypi_0    pypi
jedi                      0.18.2                   pypi_0    pypi
jinja2                    3.1.2                    pypi_0    pypi
joblib                    1.2.0                    pypi_0    pypi
jsonschema                4.17.3                   pypi_0    pypi
jupyter-client            7.4.9                    pypi_0    pypi
jupyter-core              4.12.0                   pypi_0    pypi
jupyter-server            1.23.6                   pypi_0    pypi
jupyterlab-pygments       0.2.2                    pypi_0    pypi
jupyterlab-widgets        3.0.6                    pypi_0    pypi
keras                     2.11.0                   pypi_0    pypi
kiwisolver                1.4.4                    pypi_0    pypi
ld_impl_linux-64          2.38                 h1181459_1  
libclang                  15.0.6.1                 pypi_0    pypi
libffi                    3.4.2                h6a678d5_6  
libgcc-ng                 11.2.0               h1234567_1  
libgomp                   11.2.0               h1234567_1  
libstdcxx-ng              11.2.0               h1234567_1  
linecache2                1.0.0                    pypi_0    pypi
lungmask                  0.2.13                   pypi_0    pypi
markdown                  3.3.7                    pypi_0    pypi
markupsafe                2.1.2                    pypi_0    pypi
matplotlib                3.5.3                    pypi_0    pypi
matplotlib-inline         0.1.6                    pypi_0    pypi
mergedeep                 1.3.4                    pypi_0    pypi
mistune                   2.0.5                    pypi_0    pypi
mkdocs                    1.4.2                    pypi_0    pypi
mkdocs-autorefs           0.4.1                    pypi_0    pypi
mkdocstrings              0.20.0                   pypi_0    pypi
monai                     1.1.0                    pypi_0    pypi
nbclassic                 0.5.3                    pypi_0    pypi
nbclient                  0.7.2                    pypi_0    pypi
nbconvert                 7.2.10                   pypi_0    pypi
nbformat                  5.8.0                    pypi_0    pypi
ncurses                   6.4                  h6a678d5_0  
nest-asyncio              1.5.6                    pypi_0    pypi
networkx                  2.6.3                    pypi_0    pypi
nibabel                   4.0.2                    pypi_0    pypi
notebook                  6.5.3                    pypi_0    pypi
notebook-shim             0.2.2                    pypi_0    pypi
numpy                     1.21.6                   pypi_0    pypi
nvidia-cublas-cu11        11.10.3.66               pypi_0    pypi
nvidia-cuda-nvrtc-cu11    11.7.99                  pypi_0    pypi
nvidia-cuda-runtime-cu11  11.7.99                  pypi_0    pypi
nvidia-cudnn-cu11         8.5.0.96                 pypi_0    pypi
oauthlib                  3.2.2                    pypi_0    pypi
opencv-python             4.7.0.72                 pypi_0    pypi
openssl                   1.1.1t               h7f8727e_0  
opt-einsum                3.3.0                    pypi_0    pypi
packaging                 23.0                     pypi_0    pypi
pandas                    1.3.5                    pypi_0    pypi
pandocfilters             1.5.0                    pypi_0    pypi
parso                     0.8.3                    pypi_0    pypi
patsy                     0.5.3                    pypi_0    pypi
pexpect                   4.8.0                    pypi_0    pypi
pickle-mixin              1.0.2                    pypi_0    pypi
pickleshare               0.7.5                    pypi_0    pypi
pillow                    9.4.0                    pypi_0    pypi
pip                       22.3.1           py37h06a4308_0  
pkgutil-resolve-name      1.3.10                   pypi_0    pypi
plotly                    5.13.1                   pypi_0    pypi
precision-medicine-toolbox 0.11                     pypi_0    pypi
prometheus-client         0.16.0                   pypi_0    pypi
prompt-toolkit            3.0.38                   pypi_0    pypi
protobuf                  3.19.6                   pypi_0    pypi
psutil                    5.9.4                    pypi_0    pypi
ptyprocess                0.7.0                    pypi_0    pypi
pyasn1                    0.4.8                    pypi_0    pypi
pyasn1-modules            0.2.8                    pypi_0    pypi
pycparser                 2.21                     pypi_0    pypi
pydicom                   2.3.1                    pypi_0    pypi
pygments                  2.14.0                   pypi_0    pypi
pykwalify                 1.8.0                    pypi_0    pypi
pymdown-extensions        9.10                     pypi_0    pypi
pyparsing                 3.0.9                    pypi_0    pypi
pyradiomics               3.0.1                    pypi_0    pypi
pyrsistent                0.19.3                   pypi_0    pypi
python                    3.7.16               h7a1cb2a_0  
python-dateutil           2.8.2                    pypi_0    pypi
python-gdcm               3.0.21                   pypi_0    pypi
pytz                      2022.7.1                 pypi_0    pypi
pywavelets                1.3.0                    pypi_0    pypi
pyyaml                    6.0                      pypi_0    pypi
pyyaml-env-tag            0.1                      pypi_0    pypi
pyzmq                     25.0.2                   pypi_0    pypi
readline                  8.2                  h5eee18b_0  
requests                  2.28.2                   pypi_0    pypi
requests-oauthlib         1.3.1                    pypi_0    pypi
rsa                       4.9                      pypi_0    pypi
rt-utils                  1.2.7                    pypi_0    pypi
ruamel-yaml               0.17.21                  pypi_0    pypi
ruamel-yaml-clib          0.2.7                    pypi_0    pypi
scikit-image              0.19.3                   pypi_0    pypi
scikit-learn              1.0.2                    pypi_0    pypi
scipy                     1.7.3                    pypi_0    pypi
seaborn                   0.12.2                   pypi_0    pypi
send2trash                1.8.0                    pypi_0    pypi
setuptools                65.6.3           py37h06a4308_0  
simpleitk                 2.2.1                    pypi_0    pypi
six                       1.16.0                   pypi_0    pypi
sniffio                   1.3.0                    pypi_0    pypi
soupsieve                 2.4                      pypi_0    pypi
sqlite                    3.41.1               h5eee18b_0  
statannot                 0.2.3                    pypi_0    pypi
statsmodels               0.13.5                   pypi_0    pypi
tenacity                  8.2.2                    pypi_0    pypi
tensorboard               2.11.2                   pypi_0    pypi
tensorboard-data-server   0.6.1                    pypi_0    pypi
tensorboard-plugin-wit    1.8.1                    pypi_0    pypi
tensorflow                2.11.0                   pypi_0    pypi
tensorflow-estimator      2.11.0                   pypi_0    pypi
tensorflow-io-gcs-filesystem 0.31.0                   pypi_0    pypi
termcolor                 2.2.0                    pypi_0    pypi
terminado                 0.17.1                   pypi_0    pypi
threadpoolctl             3.1.0                    pypi_0    pypi
tifffile                  2021.11.2                pypi_0    pypi
tinycss2                  1.2.1                    pypi_0    pypi
tk                        8.6.12               h1ccaba5_0  
torch                     1.13.1                   pypi_0    pypi
tornado                   6.2                      pypi_0    pypi
tqdm                      4.65.0                   pypi_0    pypi
traceback2                1.4.0                    pypi_0    pypi
traitlets                 5.9.0                    pypi_0    pypi
typing-extensions         4.5.0                    pypi_0    pypi
unittest2                 1.1.0                    pypi_0    pypi
urllib3                   1.26.15                  pypi_0    pypi
watchdog                  3.0.0                    pypi_0    pypi
wcwidth                   0.2.6                    pypi_0    pypi
webencodings              0.5.1                    pypi_0    pypi
websocket-client          1.5.1                    pypi_0    pypi
werkzeug                  2.2.3                    pypi_0    pypi
wheel                     0.38.4           py37h06a4308_0  
widgetsnbextension        4.0.6                    pypi_0    pypi
wrapt                     1.15.0                   pypi_0    pypi
xlrd                      1.2.0                    pypi_0    pypi
xz                        5.2.10               h5eee18b_1  
zipp                      3.15.0                   pypi_0    pypi
zlib                      1.2.13               h5eee18b_0

**Comment by primakov** [comment]
Thanks for sharing this! Most likely the issue is in the tensorflow version, I was very optimistic when I defined the requirements file :). Pardon the slow responses, it is a bit hard to find time to support this repo as I'm not working for the uni anymore. I'll try to fix the requirement file asap and come back to you.
